### ISSUE TITLE
Support automatic recognition of pixi environments in IDEs: VS Code

### ISSUE THREAD

**Issue by Zaubeerer** [prefix-dev/pixi#411]
### Problem description

As [discussed on Discord](https://discordapp.com/channels/1082332781146800168/1082338253925003385/1166398225259958422), there currently isn't a proper integration with VS Code yet.

It would be great if pixi environments automagically show up when selecting the Python interpreter via `Python: Select Interpreter` (see [VS Code Python Interpreter Docs](https://code.visualstudio.com/docs/python/environments#_select-and-activate-an-environment)), as do conda, poetry, venv and other environments.

### Current workaround
Manually select the pixi environment python executable as [described in the VS Code Docs](https://code.visualstudio.com/docs/python/environments#_manually-specify-an-interpreter) and mentiond in https://github.com/prefix-dev/pixi/issues/166#issuecomment-1616602232.

However, this may not work for all VS Code Python features as described in #412.

### Related information
- https://code.visualstudio.com/docs/python/environments#_how-the-extension-chooses-an-environment-automatically
- https://code.visualstudio.com/docs/python/environments#_where-the-extension-looks-for-environments

**Comment by baszalmstra** [comment]
I started some initial work here: https://github.com/microsoft/vscode-python/pull/22968

Lets see what happens!

**Comment by ruben-arts** [comment]
@j-bac is `ipykernel` in the same environments as the `python` interpeter you specify here?

**Comment by j-bac** [comment]
Yes it is part of the pixi.toml
```
[pypi-dependencies]
ipykernel = ">=6.29.3,<6.30"
```

**Comment by abkfenris** [comment]
I was able to get the IPython kernel to work in VS Code by running the kernelspec install inside the pixi environment.

```toml
[tasks]
install-kernel = "python -m ipykernel install --user --name <kernel name> --display-name <kernel name>"
```

**Comment by j-bac** [comment]
Thanks that worked!

**Comment by b4rlw** [comment]
A side effect of no auto-discovery is that if you initialise a new project and create a Jupyter notebook - VSCode won't show you any Pixi kernels before you firstly, create any `.py` file and manually select your Pixi env's Python interpreter. After a restart of the Jupyter notebook, you will be able to use that Pixi Python interpreter as a kernel without having to run any kernel creation commands manually. I assume most VSCode users will be used to this pattern by now, but thought I'd mention it incase anyone is doing that dance for the first time 😁

**Comment by roaldarbol** [comment]
I guess this issue can be closed, right? Amazing work @baszalmstra! 🥳

**Comment by ruben-arts** [comment]
Closing as it is included in the 2024.10.0 version: https://github.com/microsoft/vscode-python/releases/tag/v2024.10.0

**Comment by visr** [comment]
Question, the release notes also mention adding support for https://github.com/microsoft/python-environment-tools. This does not support Pixi: https://github.com/microsoft/python-environment-tools/issues/46. What is the difference between this and the locator? From an outsider perspective they seem to do the same.
